src/Illuminate/Database/Grammar.php

```diff
@@ -32,13 +32,13 @@ public function wrapArray(array $values)
     /**
      * Wrap a table in keyword identifiers.
      *
-     * @param  \Illuminate\Database\Query\Expression|string  $table
+     * @param  \Illuminate\Contracts\Database\Query\Expression|string  $table
      * @return string
      */
     public function wrapTable($table)
     {
         if (! $this->isExpression($table)) {
-            return $this->wrap($this->tablePrefix.$table, true);
+            return $this->wrap($this->tablePrefix.$this->getValue($table), true);
```

Comment:
If `$table` is not an expression, it is a `string` as per PHPdoc.
It looks like passing a string into `$this->getValue()` is invalid as the docblock in line 211+ states:
```
 /**
     * Get the value of a raw expression.
     *
     * @param  \Illuminate\Database\Query\Expression  $expression
     * @return mixed
     */```

If you want to take this route, you might want to allow strings for the `$expression` parameter, and probably also want to rename the parameter to something else that `$expression`, as it would be misleading.

Reply:
Good catch! Either one is incorrect by definition, but it works. I believe changing `getValue` to allow `mixed` will be the best solution as the caller does not have to check by itself of which type the value ist. Will change it soon.